runtime/altair/src/migrations.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_types::tokens::CurrencyId;
+use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
+
+use crate::{OrmlAssetRegistry, Runtime};
+
+pub type UpgradeAltair1027 = (
+	pallet_loans_ref::migrations::v1::Migration<Runtime>,
+	TrancheLocationMigration,
+);
+
+/// This migration sets the AssetMetadata.location of all the Tranche tokens
+/// registered in the AssetRegistry to `None`.
+pub struct TrancheLocationMigration;
+
+impl OnRuntimeUpgrade for TrancheLocationMigration {
+	fn on_runtime_upgrade() -> Weight {
+		for (asset_id, metadata) in orml_asset_registry::Metadata::<Runtime>::iter() {
+			if matches!(asset_id, CurrencyId::Tranche(_, _)) && metadata.location.is_some() {
+				match OrmlAssetRegistry::do_update_asset(
+					asset_id,
+					// decimals
+					None,
+					// name
+					None,
+					// symbol
+					None,
+					// existential_deposit
+					None,
+					// location: we do set it to `None`
+					Some(None),
+					// additional
+					None,
+				) {
+					Err(e) => log::error!("TrancheLocationMigration: Failed to update asset with underlying error: {:?}", e),
+					_ => continue,
+				}
+			}
+		}
+
+		// todo(nuno): not sure how to build this properly,
+		// setting it to a conservative value for now.
+		Weight::from_ref_time(200_000_000)
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<sp_std::vec::Vec<u8>, &'static str> {
+		Ok(Default::default())
```

Comment:
Could: Return list of current orml registered assets and assert in `post_upgrade` that none of these have been tampered with apart from the location, if its a tranche token.

Reply:
It's a fair recommendation but given the simplicity of this migration I will skip it at this time.